### PR TITLE
Fix issue with plotting widths

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR fixes an issue seem on fritz where the plot height was not reported properly.

### CODE REVIEW

**Comment by profjsb** [comment]
Might check that this fixes the plot sizing on mobile (which currently spills past the screen). 

Also, you might update the title of the PR since this also appears to modify the way data is grabbed in a session.

**Comment by profjsb** [comment]
Might check that this fixes the plot sizing on mobile (which currently spills past the screen). 

Also, you might update the title of the PR since this also appears to modify the way data is grabbed in a session.

**Comment by mcoughlin** [comment]
@profjsb yeah it takes care of the move to Guys new syntax. And it's really fixing a problem reported last night about the spectrum plot not showing up, which was that the width / height were not automatically set

**Comment by profjsb** [comment]
Can you check that it fixes the mobile view? Might need other small tweaks for that.

**Comment by mcoughlin** [comment]
@profjsb I am actually a bit unclear on how that works, can you summarize? :)

**Comment by mcoughlin** [comment]
@profjsb https://github.com/react-grid-layout/react-grid-layout/issues/1724 this is the core of that issue.

File: skyportal/handlers/api/internal/plot.py [hunk 1]
```diff
@@ -30,14 +30,19 @@ def get(self, obj_id):
...
-        self.verify_and_commit()
-        self.success(data={'bokehJSON': json, 'url': self.request.uri})
+        with self.Session() as session:
+            try:
+                json = plot.photometry_plot(
+                    obj_id,
+                    self.current_user.id,
+                    session,
+                    width=int(width),
+                    device=device,
```
Comment (guynir42) [inline skyportal/handlers/api/internal/plot.py:40]: Both here and in the spectroscopy plot: 
I would explicitly name each parameter I'm sending into the functions. 
There are starting to be a lot of them, and the order is hard to guess.

File: skyportal/plot.py [hunk 2]
```diff
@@ -2041,11 +2061,11 @@ def make_spectrum_layout(obj, spectra, user, device, width, smoothing, smooth_nu
         normfac = np.nanmedian(np.abs(s.fluxes))
         normfac = normfac if normfac != 0.0 else 1e-20
         altdata = json.dumps(s.altdata) if s.altdata is not None else ""
-        annotations = (
-            AnnotationOnSpectrum.query_records_accessible_by(user)
-            .filter(AnnotationOnSpectrum.spectrum_id == s.id)
-            .all()
-        )
+        annotations = session.scalars(
+            AnnotationOnSpectrum.select(session.user_or_token).where(
```
Comment (guynir42) [inline skyportal/plot.py:2065]: This is nice. You should be aware that it only works with a handler's `self.Session()` and will not work with a `DBSession()`. I don't know if that is going to be a problem or not. Probably not.

**Review by guynir42 — APPROVED** [review]
Looks good!

File: skyportal/handlers/api/internal/plot.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/internal/plot.py]: Good idea!

File: skyportal/plot.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/plot.py:2065]: Makes sense, for now we pass a handler session (which should hopefully always be the case)
